src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala

```diff
@@ -692,55 +565,101 @@ final case class ThermalGrid(
   /** Convert the given state of the thermal grid into result models of its
     * constituent models.
     *
-    * @param currentTick
-    *   Actual simulation tick.
     * @param state
-    *   State to be converted.
-    * @param startDateTime
-    *   Start date time of the simulation.
+    *   Last state of the heat pump.
+    * @param lastOperatingPoint
+    *   The last operating point of the heat pump.
+    * @param currentOperatingPoint
+    *   The actual operating point of the heat pump.
+    * @param dateTime
+    *   The actual date and time of the actual simulation tick.
     * @return
     *   A [[Seq]] of results of the constituent thermal model.
     */
-  def results(currentTick: Long, state: ThermalGridState)(implicit
-      startDateTime: ZonedDateTime
+  def results(
+      state: HpState,
+      lastOperatingPoint: Option[HpOperatingPoint],
+      currentOperatingPoint: HpOperatingPoint,
+      dateTime: ZonedDateTime,
   ): Seq[ResultEntity] = {
 
-    val maybeHouseResult = house
-      .zip(state.houseState)
-      .filter { case (_, state) => state.tick == currentTick }
-      .map {
-        case (
-              thermalHouse,
-              ThermalHouseState(tick, innerTemperature, thermalInfeed),
-            ) =>
+    val currentOpThermals = currentOperatingPoint.thermalOps
+
+    val lastOpThermals = lastOperatingPoint match {
+      case Some(op) => op.thermalOps
+      case None     =>
+        // we need some thermals that are different from zero for the first result
+        ThermalOpWrapper(Kilowatts(-42), Kilowatts(-42), Kilowatts(-42))
+    }
```

Comment:
This could be solved by making lastOpThermals optional (you then need the further adaptions below):

```suggestion
    val lastOpThermals = lastOperatingPoint.map(_.thermalOps)
```